### DISCUSSION TITLE
Does helm have an equivalent of ivy-rich

### DISCUSSION THREAD

**Discussion by rbenson** [emacs-helm/helm#2490] [Q&A] [answered]
Hi, I have been watching videos, and after some poking around, found people have beefed up their experience by installing `ivy-rich`.  Is there something equivalent in helm, particularly for `helm-M-x`.

**Comment by thierryvolpiatto** [comment, marked as answer]
Ryan B ***@***.***> writes:

> 1.  ( ) text/plain          (*) text/html           
>
> Hi, I have been watching videos, and after some poking around, found
> people have beefed up their experience by installing ivy-rich. Is
> there something equivalent in helm, particularly for helm-M-x.

I don't know what is ivy-rich and what it does.
What do you miss in helm-M-x?

-- 
Thierry

**Reply by rbenson** [reply]
Sorry for the delay, i didn't get a notification. It adds extra details to in M-x and such, see https://github.com/Yevgnen/ivy-rich/blob/master/screenshots.org

**Reply by thierryvolpiatto** [reply]
Now you can hit `C-]` in helm-M-x.

**Reply by rbenson** [reply]
Sweet - thanks.

**Comment by dmitrym0** [comment]
Hey folks,

Wondering about the same. When I hit `helm-m-x`, I don't get any doc strings in the resulting window:

<img width="565" alt="image" src="https://user-images.githubusercontent.com/707020/178634280-67a4af49-5414-4fac-99bd-57ab3ca1fdd8.png">

What am I missing? `C-]` is not a binding I can find.

**Reply by thierryvolpiatto** [reply]
Looks what you are using is a modified Emacs or Helm, Spacemacs perhaps? Generally all problems come from there.
Here a vanilla Helm after hitting C-]:
![Capture d’écran_2022-06-14_07-12-25](https://user-images.githubusercontent.com/1533939/178647863-05335e45-e9d3-4b64-a2e7-a73a96476342.png)

**Reply by dmitrym0** [reply]
@thierryvolpiatto Vanilla emacs on OS X 🤷 

I don't have anything bound to `C-]`. Do you mind doing a `describe-key` on your instance?

**Comment by thierryvolpiatto** [comment]
Dmitry ***@***.***> writes:

> @thierryvolpiatto Vanilla emacs on OS X 🤷
>
> I don't have anything bound to C-]. Do you mind doing a describe-key on your instance?

You have it, just it have been disabled, check your config.

    (defvar helm-M-x-map
      (let ((map (make-sparse-keymap)))
        (set-keymap-parent map helm-comp-read-map)
        (define-key map (kbd "C-u") nil)
        (define-key map (kbd "C-u") #'helm-M-x-universal-argument)
        (define-key map (kbd "C-]") #'helm-M-x-toggle-short-doc)
        map))


C-h k from helm-M-x:

    C-] runs the command helm-M-x-toggle-short-doc, which is an
    interactive compiled Lisp function in `helm-command.el'.

    It is bound to C-].

    (helm-M-x-toggle-short-doc)

    Toggle short doc display in helm-M-x.

I forgot adding the command from helm-M-x-map in helm-M-x help (C-h m),
will do soon.

-- 
Thierry

**Comment by dmitrym0** [comment]
```
eval: Symbol’s function definition is void: helm-M-x-toggle-short-doc
```

and
 
```
~/.emacs.d ❯❯❯ grep -ri 'helm-M-x-toggle-short-doc' *
~/.emacs.d ❯❯❯
```

I think I might have a different problem! Thanks for letting me know what the command should be.

**Reply by dmitrym0** [reply]
Looks like I was using an older version of Helm. With an upgrade it's there. Thanks @thierryvolpiatto
